frontend/ui/screensaver.lua

```diff
@@ -449,7 +449,27 @@ function Screensaver:setup(event, event_message)
             else
                 doc_settings = DocSettings:open(lastfile)
             end
-            excluded = doc_settings:isTrue("exclude_screensaver")
+
+            local exclude_finished_books = G_reader_settings:isTrue("screensaver_exclude_finished_books")
+            local exclude_on_hold_books = G_reader_settings:isTrue("screensaver_exclude_on_hold_books")
+            local exclude_screensaver_fm = G_reader_settings:isTrue("screensaver_hide_cover_in_filemanager")
+            local book_summary = doc_settings:readSetting("summary")
+            local book_finished_or_on_hold = book_summary and (book_summary.status == "complete" or book_summary.status == "abandoned")
+            local exclude_screensaver_finished_or_on_hold = false
+            if (exclude_finished_books or exclude_on_hold_books) and book_finished_or_on_hold then
+                exclude_screensaver_finished_or_on_hold = true
+                if self.show_message then
+                    self.show_message = false
+                end
+            end
+            if ui then -- ReaderUI instance
+                excluded = doc_settings:isTrue("exclude_screensaver") or exclude_screensaver_finished_or_on_hold
+            else -- FileManager instance
+                excluded = doc_settings:isTrue("exclude_screensaver") or exclude_screensaver_finished_or_on_hold or exclude_screensaver_fm
+                if self.show_message and exclude_screensaver_fm then
+                    self.show_message = false
+                end
```

Comment:
All this logic is tedious to read - and because of that I think there is a bug (see other comment).
You should write it simpler and more logical
```lua
excluded = doc_settings:isTrue("exclude_screensaver")
if not excluded then
    if not ui and G_reader_settings:isTrue("screensaver_hide_cover_in_filemanager") then
        excluded = true
    else if ui then
        if  G_reader_settings:isTrue("screensaver_exclude_finished_books") or G_reader_settings:isTrue("screensaver_exclude_on_hold_books") then
            -- do your summary cuisine and checks and set exclude = true
        end
    end
end
if excluded then
    self.show_message = false
end
```